diff --git a/include/afl-as.h b/include/afl-as.h
index c005d43d..20c09fcc 100644
--- a/include/afl-as.h
+++ b/include/afl-as.h
@@ -421,7 +421,7 @@ static const u8 *main_payload_64 =
   "\n"
   "__afl_maybe_log:\n"
   "\n"
-#if defined(__OpenBSD__) || (defined(__FreeBSD__) && (__FreeBSD__ < 9))
+#if defined(__OpenBSD__)
   "  .byte 0x9f /* lahf */\n"
 #else
   "  lahf\n"
diff --git a/include/afl-fuzz.h b/include/afl-fuzz.h
index 34b626f9..b78dd155 100644
--- a/include/afl-fuzz.h
+++ b/include/afl-fuzz.h
@@ -76,10 +76,10 @@
 #include <sys/file.h>
 #include <sys/types.h>
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
+#if defined(__APPLE__) || defined(__OpenBSD__) || \
     defined(__NetBSD__) || defined(__DragonFly__)
   #include <sys/sysctl.h>
-#endif                           /* __APPLE__ || __FreeBSD__ || __OpenBSD__ */
+#endif                           /* __APPLE__ || __OpenBSD__ */
 
 #if defined(__HAIKU__)
   #include <kernel/OS.h>
@@ -89,14 +89,11 @@
 /* For systems that have sched_setaffinity; right now just Linux, but one
    can hope... */
 
-#if defined(__linux__) || defined(__FreeBSD__) || defined(__NetBSD__) || \
+#if defined(__linux__) || defined(__NetBSD__) || \
     defined(__DragonFly__) || defined(__sun)
   #define HAVE_AFFINITY 1
-  #if defined(__FreeBSD__) || defined(__DragonFly__)
+  #if defined(__DragonFly__)
     #include <sys/param.h>
-    #if defined(__FreeBSD__)
-      #include <sys/cpuset.h>
-    #endif
     #include <sys/user.h>
     #include <pthread.h>
     #include <pthread_np.h>
diff --git a/src/afl-fuzz-init.c b/src/afl-fuzz-init.c
index a2fc4528..adcf6529 100644
--- a/src/afl-fuzz-init.c
+++ b/src/afl-fuzz-init.c
@@ -35,7 +35,7 @@
 
 static u8 bind_cpu(afl_state_t *afl, s32 cpuid) {
 
-  #if defined(__linux__) || defined(__FreeBSD__) || defined(__DragonFly__)
+  #if defined(__linux__) || defined(__DragonFly__)
   cpu_set_t c;
   #elif defined(__NetBSD__)
   cpuset_t *c;
@@ -45,7 +45,7 @@ static u8 bind_cpu(afl_state_t *afl, s32 cpuid) {
 
   afl->cpu_aff = cpuid;
 
-  #if defined(__linux__) || defined(__FreeBSD__) || defined(__DragonFly__)
+  #if defined(__linux__) || defined(__DragonFly__)
 
   CPU_ZERO(&c);
   CPU_SET(cpuid, &c);
@@ -67,7 +67,7 @@ static u8 bind_cpu(afl_state_t *afl, s32 cpuid) {
 
   return (sched_setaffinity(0, sizeof(c), &c) == 0);
 
-  #elif defined(__FreeBSD__) || defined(__DragonFly__)
+  #elif defined(__DragonFly__)
 
   return (pthread_setaffinity_np(pthread_self(), sizeof(c), &c) == 0);
 
@@ -254,7 +254,7 @@ void bind_to_free_cpu(afl_state_t *afl) {
 
   closedir(d);
 
-  #elif defined(__FreeBSD__) || defined(__DragonFly__)
+  #elif defined(__DragonFly__)
 
   struct kinfo_proc *procs;
   size_t             nprocs;
@@ -283,19 +283,7 @@ void bind_to_free_cpu(afl_state_t *afl) {
 
   for (i = 0; i < (s32)proccount; i++) {
 
-    #if defined(__FreeBSD__)
-
-    if (!strcmp(procs[i].ki_comm, "idle")) continue;
-
-    // fix when ki_oncpu = -1
-    s32 oncpu;
-    oncpu = procs[i].ki_oncpu;
-    if (oncpu == -1) oncpu = procs[i].ki_lastcpu;
-
-    if (oncpu != -1 && oncpu < (s32)sizeof(cpu_used) && procs[i].ki_pctcpu > 60)
-      cpu_used[oncpu] = 1;
-
-    #elif defined(__DragonFly__)
+    #if defined(__DragonFly__)
 
     if (procs[i].kp_lwp.kl_cpuid < (s32)sizeof(cpu_used) &&
         procs[i].kp_lwp.kl_pctcpu > 10)
@@ -1709,7 +1697,7 @@ double get_runnable_processes(void) {
 
   double res = 0;
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
+#if defined(__APPLE__) || defined(__OpenBSD__) || \
     defined(__NetBSD__) || defined(__DragonFly__)
 
   /* I don't see any portable sysctl or so that would quickly give us the
@@ -1754,7 +1742,7 @@ double get_runnable_processes(void) {
 
   }
 
-#endif          /* ^(__APPLE__ || __FreeBSD__ || __OpenBSD__ || __NetBSD__) */
+#endif          /* ^(__APPLE__ || __OpenBSD__ || __NetBSD__) */
 
   return res;
 
@@ -2585,7 +2573,7 @@ void check_cpu_governor(afl_state_t *afl) {
 
 void get_core_count(afl_state_t *afl) {
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
+#if defined(__APPLE__) || defined(__OpenBSD__) || \
     defined(__DragonFly__)
 
   size_t s = sizeof(afl->cpu_core_count);
@@ -2625,7 +2613,7 @@ void get_core_count(afl_state_t *afl) {
 
   #endif                                                  /* ^HAVE_AFFINITY */
 
-#endif                        /* ^(__APPLE__ || __FreeBSD__ || __OpenBSD__) */
+#endif                        /* ^(__APPLE__ || __OpenBSD__) */
 
   if (afl->cpu_core_count > 0) {
 
@@ -2633,14 +2621,14 @@ void get_core_count(afl_state_t *afl) {
 
     cur_runnable = (u32)get_runnable_processes();
 
-#if defined(__APPLE__) || defined(__FreeBSD__) || defined(__OpenBSD__) || \
+#if defined(__APPLE__) || defined(__OpenBSD__) || \
     defined(__DragonFly__)
 
     /* Add ourselves, since the 1-minute average doesn't include that yet. */
 
     ++cur_runnable;
 
-#endif                           /* __APPLE__ || __FreeBSD__ || __OpenBSD__ */
+#endif                           /* __APPLE__ || __OpenBSD__ */
 
     OKF("You have %d CPU core%s and %u runnable tasks (utilization: %0.0f%%).",
         afl->cpu_core_count, afl->cpu_core_count > 1 ? "s" : "", cur_runnable,
