diff --git a/src/afl-common.c b/src/afl-common.c
index efdb5d60..5fd41120 100644
--- a/src/afl-common.c
+++ b/src/afl-common.c
@@ -537,14 +537,13 @@ u8 *find_afl_binary(u8 *own_loc, u8 *fname) {
 
   if (perm == X_OK) {
 
-    target_path = alloc_printf("%s/%s", BIN_PATH, fname);
-
-  } else {
-
-    target_path = alloc_printf("%s/%s", AFL_PATH, fname);
+    WARNF("-DBIN_PATH not set. Will be ignored. Setting AFL_PATH instead.");
+//    target_path = alloc_printf("%s/%s", BIN_PATH, fname);
 
   }
 
+  target_path = alloc_printf("%s/%s", AFL_PATH, fname);
+
   if (!access(target_path, perm)) {
 
     return target_path;
diff --git a/src/afl-fuzz.c b/src/afl-fuzz.c
index 5a1150ce..6ea5f26f 100644
--- a/src/afl-fuzz.c
+++ b/src/afl-fuzz.c
@@ -496,7 +496,7 @@ int main(int argc, char **argv_orig, char **envp) {
   struct timeval  tv;
   struct timezone tz;
 
-  doc_path = access(DOC_PATH, F_OK) != 0 ? (u8 *)"docs" : (u8 *)DOC_PATH;
+  doc_path = (u8 *)"docs";
 
   if (argc > 1 && strcmp(argv_orig[1], "--version") == 0) {
 
