diff --git a/include/forkserver.h b/include/forkserver.h
index 593e34a2..e7c67a15 100644
--- a/include/forkserver.h
+++ b/include/forkserver.h
@@ -206,6 +206,10 @@ typedef struct afl_forkserver {
   s32                   nyx_log_fd;
 #endif
 
+  // extra values to provide the forkserver with the necessary ids to request a shared memory segment
+  s32 shm_coverage_map_shmid;
+  s32 shm_fuzzing_shmid;
+
 } afl_forkserver_t;
 
 typedef enum fsrv_run_result {
diff --git a/src/afl-common.c b/src/afl-common.c
index efdb5d60..586e0988 100644
--- a/src/afl-common.c
+++ b/src/afl-common.c
@@ -1413,52 +1413,3 @@ s32 create_file(u8 *fn) {
 
 }
 
-#ifdef __linux__
-
-/* Nyx requires a tmp workdir to access specific files (such as mmapped files,
- * etc.). This helper function basically creates both a path to a tmp workdir
- * and the workdir itself. If the environment variable TMPDIR is set, we use
- * that as the base directory, otherwise we use /tmp. */
-char *create_nyx_tmp_workdir(void) {
-
-  char *tmpdir = getenv("TMPDIR");
-
-  if (!tmpdir) { tmpdir = "/tmp"; }
-
-  char *nyx_out_dir_path =
-      alloc_printf("%s/.nyx_tmp_%d/", tmpdir, (u32)getpid());
-
-  if (mkdir(nyx_out_dir_path, 0700)) { PFATAL("Unable to create nyx workdir"); }
-
-  return nyx_out_dir_path;
-
-}
-
-/* Vice versa, we remove the tmp workdir for nyx with this helper function. */
-void remove_nyx_tmp_workdir(afl_forkserver_t *fsrv, char *nyx_out_dir_path) {
-
-  char *workdir_path = alloc_printf("%s/workdir", nyx_out_dir_path);
-
-  if (access(workdir_path, R_OK) == 0) {
-
-    if (fsrv->nyx_handlers->nyx_remove_work_dir(workdir_path) != true) {
-
-      WARNF("Unable to remove nyx workdir (%s)", workdir_path);
-
-    }
-
-  }
-
-  if (rmdir(nyx_out_dir_path)) {
-
-    WARNF("Unable to remove nyx workdir (%s)", nyx_out_dir_path);
-
-  }
-
-  ck_free(workdir_path);
-  ck_free(nyx_out_dir_path);
-
-}
-
-#endif
-
diff --git a/src/afl-forkserver.c b/src/afl-forkserver.c
index 71d8570d..ed526d91 100644
--- a/src/afl-forkserver.c
+++ b/src/afl-forkserver.c
@@ -52,139 +52,7 @@
 #include <sys/select.h>
 #include <sys/stat.h>
 
-#ifdef __linux__
-  #include <dlfcn.h>
-
-/* function to load nyx_helper function from libnyx.so */
-
-nyx_plugin_handler_t *afl_load_libnyx_plugin(u8 *libnyx_binary) {
-
-  void                 *handle;
-  nyx_plugin_handler_t *plugin = calloc(1, sizeof(nyx_plugin_handler_t));
-
-  ACTF("Trying to load libnyx.so plugin...");
-  handle = dlopen((char *)libnyx_binary, RTLD_NOW);
-  if (!handle) { goto fail; }
-
-  plugin->nyx_config_load = dlsym(handle, "nyx_config_load");
-  if (plugin->nyx_config_load == NULL) { goto fail; }
-
-  plugin->nyx_config_set_workdir_path =
-      dlsym(handle, "nyx_config_set_workdir_path");
-  if (plugin->nyx_config_set_workdir_path == NULL) { goto fail; }
-
-  plugin->nyx_config_set_input_buffer_size =
-      dlsym(handle, "nyx_config_set_input_buffer_size");
-  if (plugin->nyx_config_set_input_buffer_size == NULL) { goto fail; }
-
-  plugin->nyx_config_set_input_buffer_write_protection =
-      dlsym(handle, "nyx_config_set_input_buffer_write_protection");
-  if (plugin->nyx_config_set_input_buffer_write_protection == NULL) {
-
-    goto fail;
-
-  }
-
-  plugin->nyx_config_set_hprintf_fd =
-      dlsym(handle, "nyx_config_set_hprintf_fd");
-  if (plugin->nyx_config_set_hprintf_fd == NULL) { goto fail; }
-
-  plugin->nyx_config_set_process_role =
-      dlsym(handle, "nyx_config_set_process_role");
-  if (plugin->nyx_config_set_process_role == NULL) { goto fail; }
-
-  plugin->nyx_config_set_reuse_snapshot_path =
-      dlsym(handle, "nyx_config_set_reuse_snapshot_path");
-  if (plugin->nyx_config_set_reuse_snapshot_path == NULL) { goto fail; }
-
-  plugin->nyx_new = dlsym(handle, "nyx_new");
-  if (plugin->nyx_new == NULL) { goto fail; }
-
-  plugin->nyx_shutdown = dlsym(handle, "nyx_shutdown");
-  if (plugin->nyx_shutdown == NULL) { goto fail; }
-
-  plugin->nyx_option_set_reload_mode =
-      dlsym(handle, "nyx_option_set_reload_mode");
-  if (plugin->nyx_option_set_reload_mode == NULL) { goto fail; }
-
-  plugin->nyx_option_set_timeout = dlsym(handle, "nyx_option_set_timeout");
-  if (plugin->nyx_option_set_timeout == NULL) { goto fail; }
-
-  plugin->nyx_option_apply = dlsym(handle, "nyx_option_apply");
-  if (plugin->nyx_option_apply == NULL) { goto fail; }
-
-  plugin->nyx_set_afl_input = dlsym(handle, "nyx_set_afl_input");
-  if (plugin->nyx_set_afl_input == NULL) { goto fail; }
-
-  plugin->nyx_exec = dlsym(handle, "nyx_exec");
-  if (plugin->nyx_exec == NULL) { goto fail; }
-
-  plugin->nyx_get_bitmap_buffer = dlsym(handle, "nyx_get_bitmap_buffer");
-  if (plugin->nyx_get_bitmap_buffer == NULL) { goto fail; }
-
-  plugin->nyx_get_bitmap_buffer_size =
-      dlsym(handle, "nyx_get_bitmap_buffer_size");
-  if (plugin->nyx_get_bitmap_buffer_size == NULL) { goto fail; }
-
-  plugin->nyx_get_aux_string = dlsym(handle, "nyx_get_aux_string");
-  if (plugin->nyx_get_aux_string == NULL) { goto fail; }
-
-  plugin->nyx_remove_work_dir = dlsym(handle, "nyx_remove_work_dir");
-  if (plugin->nyx_remove_work_dir == NULL) { goto fail; }
-
-  plugin->nyx_config_set_aux_buffer_size =
-      dlsym(handle, "nyx_config_set_aux_buffer_size");
-  if (plugin->nyx_config_set_aux_buffer_size == NULL) { goto fail; }
-
-  OKF("libnyx plugin is ready!");
-  return plugin;
-
-fail:
-
-  FATAL("failed to load libnyx: %s\n", dlerror());
-  ck_free(plugin);
-  return NULL;
-
-}
-
-void afl_nyx_runner_kill(afl_forkserver_t *fsrv) {
-
-  if (fsrv->nyx_mode) {
-
-    if (fsrv->nyx_aux_string) { ck_free(fsrv->nyx_aux_string); }
-
-    /* check if we actually got a valid nyx runner */
-    if (fsrv->nyx_runner) {
-
-      fsrv->nyx_handlers->nyx_shutdown(fsrv->nyx_runner);
-
-    }
-
-    /* if we have use a tmp work dir we need to remove it */
-    if (fsrv->nyx_use_tmp_workdir && fsrv->nyx_tmp_workdir_path) {
-
-      remove_nyx_tmp_workdir(fsrv, fsrv->nyx_tmp_workdir_path);
-
-    }
-
-    if (fsrv->nyx_log_fd >= 0) { close(fsrv->nyx_log_fd); }
-
-  }
-
-}
-
-  /* Wrapper for FATAL() that kills the nyx runner (and removes all created tmp
-   * files) before exiting. Used before "afl_fsrv_killall()" is registered as
-   * an atexit() handler. */
-  #define NYX_PRE_FATAL(fsrv, x...) \
-    do {                            \
-                                    \
-      afl_nyx_runner_kill(fsrv);    \
-      FATAL(x);                     \
-                                    \
-    } while (0)
-
-#endif
+#include "forkserver_wrapper.h"
 
 /**
  * The correct fds for reading and writing pipes
@@ -531,276 +399,6 @@ void afl_fsrv_start(afl_forkserver_t *fsrv, char **argv,
   s32   rlen;
   char *ignore_autodict = getenv("AFL_NO_AUTODICT");
 
-#ifdef __linux__
-  if (unlikely(fsrv->nyx_mode)) {
-
-    if (fsrv->nyx_runner != NULL) { return; }
-
-    if (!be_quiet) { ACTF("Spinning up the NYX backend..."); }
-
-    if (fsrv->nyx_use_tmp_workdir) {
-
-      fsrv->nyx_tmp_workdir_path = create_nyx_tmp_workdir();
-      fsrv->out_dir_path = fsrv->nyx_tmp_workdir_path;
-
-    } else {
-
-      if (fsrv->out_dir_path == NULL) {
-
-        NYX_PRE_FATAL(fsrv, "Nyx workdir path not found...");
-
-      }
-
-    }
-
-    /* libnyx expects an absolute path */
-    char *outdir_path_absolute = realpath(fsrv->out_dir_path, NULL);
-    if (outdir_path_absolute == NULL) {
-
-      NYX_PRE_FATAL(fsrv, "Nyx workdir path cannot be resolved ...");
-
-    }
-
-    char *workdir_path = alloc_printf("%s/workdir", outdir_path_absolute);
-
-    if (fsrv->nyx_id == 0xFFFFFFFF) {
-
-      NYX_PRE_FATAL(fsrv, "Nyx ID is not set...");
-
-    }
-
-    if (fsrv->nyx_bind_cpu_id == 0xFFFFFFFF) {
-
-      NYX_PRE_FATAL(fsrv, "Nyx CPU ID is not set...");
-
-    }
-
-    void *nyx_config = fsrv->nyx_handlers->nyx_config_load(fsrv->target_path);
-
-    fsrv->nyx_handlers->nyx_config_set_workdir_path(nyx_config, workdir_path);
-    fsrv->nyx_handlers->nyx_config_set_input_buffer_size(nyx_config,
-                                                         fsrv->max_length);
-    fsrv->nyx_handlers->nyx_config_set_input_buffer_write_protection(nyx_config,
-                                                                     true);
-
-    char *nyx_log_path = getenv("AFL_NYX_LOG");
-    if (nyx_log_path) {
-
-      fsrv->nyx_log_fd =
-          open(nyx_log_path, O_CREAT | O_TRUNC | O_WRONLY, DEFAULT_PERMISSION);
-      if (fsrv->nyx_log_fd < 0) {
-
-        NYX_PRE_FATAL(fsrv, "AFL_NYX_LOG path could not be written");
-
-      }
-
-      fsrv->nyx_handlers->nyx_config_set_hprintf_fd(nyx_config,
-                                                    fsrv->nyx_log_fd);
-
-    }
-
-    if (fsrv->nyx_standalone) {
-
-      fsrv->nyx_handlers->nyx_config_set_process_role(nyx_config, StandAlone);
-
-    } else {
-
-      if (fsrv->nyx_parent) {
-
-        fsrv->nyx_handlers->nyx_config_set_process_role(nyx_config, Parent);
-
-      } else {
-
-        fsrv->nyx_handlers->nyx_config_set_process_role(nyx_config, Child);
-
-      }
-
-    }
-
-    if (getenv("AFL_NYX_AUX_SIZE") != NULL) {
-
-      fsrv->nyx_aux_string_len = atoi(getenv("AFL_NYX_AUX_SIZE"));
-
-      if (fsrv->nyx_handlers->nyx_config_set_aux_buffer_size(
-              nyx_config, fsrv->nyx_aux_string_len) != 1) {
-
-        NYX_PRE_FATAL(fsrv,
-                      "Invalid AFL_NYX_AUX_SIZE value set (must be a multiple "
-                      "of 4096) ...");
-
-      }
-
-    } else {
-
-      fsrv->nyx_aux_string_len = 0x1000;
-
-    }
-
-    if (getenv("AFL_NYX_REUSE_SNAPSHOT") != NULL) {
-
-      if (access(getenv("AFL_NYX_REUSE_SNAPSHOT"), F_OK) == -1) {
-
-        NYX_PRE_FATAL(fsrv, "AFL_NYX_REUSE_SNAPSHOT path does not exist");
-
-      }
-
-      /* stupid sanity check to avoid passing an empty or invalid snapshot
-       * directory */
-      char *snapshot_file_path =
-          alloc_printf("%s/global.state", getenv("AFL_NYX_REUSE_SNAPSHOT"));
-      if (access(snapshot_file_path, R_OK) == -1) {
-
-        NYX_PRE_FATAL(fsrv,
-                      "AFL_NYX_REUSE_SNAPSHOT path does not contain a valid "
-                      "Nyx snapshot");
-
-      }
-
-      ck_free(snapshot_file_path);
-
-      /* another sanity check to avoid passing a snapshot directory that is
-       * located in the current workdir (the workdir will be wiped by libnyx on
-       * startup) */
-      char *workdir_snapshot_path =
-          alloc_printf("%s/workdir/snapshot", outdir_path_absolute);
-      char *reuse_snapshot_path_real =
-          realpath(getenv("AFL_NYX_REUSE_SNAPSHOT"), NULL);
-
-      if (strcmp(workdir_snapshot_path, reuse_snapshot_path_real) == 0) {
-
-        NYX_PRE_FATAL(
-            fsrv,
-            "AFL_NYX_REUSE_SNAPSHOT path is located in current workdir "
-            "(use another output directory)");
-
-      }
-
-      ck_free(reuse_snapshot_path_real);
-      ck_free(workdir_snapshot_path);
-
-      fsrv->nyx_handlers->nyx_config_set_reuse_snapshot_path(
-          nyx_config, getenv("AFL_NYX_REUSE_SNAPSHOT"));
-
-    }
-
-    fsrv->nyx_runner = fsrv->nyx_handlers->nyx_new(nyx_config, fsrv->nyx_id);
-
-    ck_free(workdir_path);
-    ck_free(outdir_path_absolute);
-
-    if (fsrv->nyx_runner == NULL) { FATAL("Something went wrong ..."); }
-
-    u32 tmp_map_size =
-        fsrv->nyx_handlers->nyx_get_bitmap_buffer_size(fsrv->nyx_runner);
-    fsrv->real_map_size = tmp_map_size;
-    fsrv->map_size = (((tmp_map_size + 63) >> 6) << 6);
-    if (!be_quiet) { ACTF("Target map size: %u", fsrv->real_map_size); }
-
-    fsrv->trace_bits =
-        fsrv->nyx_handlers->nyx_get_bitmap_buffer(fsrv->nyx_runner);
-
-    fsrv->nyx_handlers->nyx_option_set_reload_mode(
-        fsrv->nyx_runner, getenv("AFL_NYX_DISABLE_SNAPSHOT_MODE") == NULL);
-    fsrv->nyx_handlers->nyx_option_apply(fsrv->nyx_runner);
-
-    fsrv->nyx_handlers->nyx_option_set_timeout(fsrv->nyx_runner, 2, 0);
-    fsrv->nyx_handlers->nyx_option_apply(fsrv->nyx_runner);
-
-    fsrv->nyx_aux_string = malloc(fsrv->nyx_aux_string_len);
-    memset(fsrv->nyx_aux_string, 0, fsrv->nyx_aux_string_len);
-
-    /* dry run */
-    fsrv->nyx_handlers->nyx_set_afl_input(fsrv->nyx_runner, "INIT", 4);
-    switch (fsrv->nyx_handlers->nyx_exec(fsrv->nyx_runner)) {
-
-      case Abort:
-        NYX_PRE_FATAL(fsrv, "Error: Nyx abort occurred...");
-        break;
-      case IoError:
-        NYX_PRE_FATAL(fsrv, "Error: QEMU-Nyx has died...");
-        break;
-      case Error:
-        NYX_PRE_FATAL(fsrv, "Error: Nyx runtime error has occurred...");
-        break;
-      default:
-        break;
-
-    }
-
-    /* autodict in Nyx mode */
-    if (!ignore_autodict && fsrv->add_extra_func) {
-
-      char *x =
-          alloc_printf("%s/workdir/dump/afl_autodict.txt", fsrv->out_dir_path);
-      int nyx_autodict_fd = open(x, O_RDONLY);
-      ck_free(x);
-
-      if (nyx_autodict_fd >= 0) {
-
-        struct stat st;
-        if (fstat(nyx_autodict_fd, &st) >= 0) {
-
-          u32 f_len = st.st_size;
-          u8 *dict = ck_alloc(f_len);
-          if (dict == NULL) {
-
-            NYX_PRE_FATAL(
-                fsrv, "Could not allocate %u bytes of autodictionary memory",
-                f_len);
-
-          }
-
-          u32 offset = 0, count = 0;
-          u32 len = f_len;
-
-          while (len != 0) {
-
-            rlen = read(nyx_autodict_fd, dict + offset, len);
-            if (rlen > 0) {
-
-              len -= rlen;
-              offset += rlen;
-
-            } else {
-
-              NYX_PRE_FATAL(
-                  fsrv,
-                  "Reading autodictionary fail at position %u with %u bytes "
-                  "left.",
-                  offset, len);
-
-            }
-
-          }
-
-          offset = 0;
-          while (offset < (u32)f_len &&
-                 (u8)dict[offset] + offset < (u32)f_len) {
-
-            fsrv->add_extra_func(fsrv->afl_ptr, dict + offset + 1,
-                                 (u8)dict[offset]);
-            offset += (1 + dict[offset]);
-            count++;
-
-          }
-
-          if (!be_quiet) { ACTF("Loaded %u autodictionary entries", count); }
-          ck_free(dict);
-
-        }
-
-        close(nyx_autodict_fd);
-
-      }
-
-    }
-
-    return;
-
-  }
-
-#endif
-
   if (!be_quiet) { ACTF("Spinning up the fork server..."); }
 
 #ifdef AFL_PERSISTENT_RECORD
@@ -839,124 +437,8 @@ void afl_fsrv_start(afl_forkserver_t *fsrv, char **argv,
   if (pipe(st_pipe) || pipe(ctl_pipe)) { PFATAL("pipe() failed"); }
 
   fsrv->last_run_timed_out = 0;
-  fsrv->fsrv_pid = fork();
-
-  if (fsrv->fsrv_pid < 0) { PFATAL("fork() failed"); }
-
-  if (!fsrv->fsrv_pid) {
-
-    /* CHILD PROCESS */
-
-    // enable terminating on sigpipe in the childs
-    struct sigaction sa;
-    memset((char *)&sa, 0, sizeof(sa));
-    sa.sa_handler = SIG_DFL;
-    sigaction(SIGPIPE, &sa, NULL);
-
-    struct rlimit r;
-
-    if (!fsrv->cmplog_binary) {
-
-      unsetenv(CMPLOG_SHM_ENV_VAR);  // we do not want that in non-cmplog fsrv
-
-    }
-
-    /* Umpf. On OpenBSD, the default fd limit for root users is set to
-       soft 128. Let's try to fix that... */
-    if (!getrlimit(RLIMIT_NOFILE, &r) && r.rlim_cur < FORKSRV_FD + 2) {
-
-      r.rlim_cur = FORKSRV_FD + 2;
-      setrlimit(RLIMIT_NOFILE, &r);                        /* Ignore errors */
-
-    }
-
-    if (fsrv->mem_limit) {
-
-      r.rlim_max = r.rlim_cur = ((rlim_t)fsrv->mem_limit) << 20;
-
-#ifdef RLIMIT_AS
-      setrlimit(RLIMIT_AS, &r);                            /* Ignore errors */
-#else
-      /* This takes care of OpenBSD, which doesn't have RLIMIT_AS, but
-         according to reliable sources, RLIMIT_DATA covers anonymous
-         maps - so we should be getting good protection against OOM bugs. */
-
-      setrlimit(RLIMIT_DATA, &r);                          /* Ignore errors */
-#endif                                                        /* ^RLIMIT_AS */
-
-    }
-
-    /* Dumping cores is slow and can lead to anomalies if SIGKILL is delivered
-       before the dump is complete. */
-
-    if (!fsrv->debug) {
-
-      r.rlim_max = r.rlim_cur = 0;
-      setrlimit(RLIMIT_CORE, &r);                          /* Ignore errors */
-
-    }
-
-    /* Isolate the process and configure standard descriptors. If out_file is
-       specified, stdin is /dev/null; otherwise, out_fd is cloned instead. */
-
-    setsid();
 
-    if (!(debug_child_output)) {
-
-      dup2(fsrv->dev_null_fd, 1);
-      dup2(fsrv->dev_null_fd, 2);
-
-    }
-
-    if (!fsrv->use_stdin) {
-
-      dup2(fsrv->dev_null_fd, 0);
-
-    } else {
-
-      dup2(fsrv->out_fd, 0);
-      close(fsrv->out_fd);
-
-    }
-
-    /* Set up control and status pipes, close the unneeded original fds. */
-
-    if (dup2(ctl_pipe[0], FORKSRV_FD) < 0) { PFATAL("dup2() failed"); }
-    if (dup2(st_pipe[1], FORKSRV_FD + 1) < 0) { PFATAL("dup2() failed"); }
-
-    close(ctl_pipe[0]);
-    close(ctl_pipe[1]);
-    close(st_pipe[0]);
-    close(st_pipe[1]);
-
-    close(fsrv->out_dir_fd);
-    close(fsrv->dev_null_fd);
-    close(fsrv->dev_urandom_fd);
-
-    if (fsrv->plot_file != NULL) {
-
-      fclose(fsrv->plot_file);
-      fsrv->plot_file = NULL;
-
-    }
-
-    /* This should improve performance a bit, since it stops the linker from
-       doing extra work post-fork(). */
-
-    if (!getenv("LD_BIND_LAZY")) { setenv("LD_BIND_NOW", "1", 1); }
-
-    /* Set sane defaults for sanitizers */
-    set_sanitizer_defaults();
-
-    fsrv->init_child_func(fsrv, argv);
-
-    /* Use a distinctive bitmap signature to tell the parent about execv()
-       falling through. */
-
-    *(u32 *)fsrv->trace_bits = EXEC_FAIL_SIG;
-    FATAL("Error: execv to target failed\n");
-
-  }
+  call_report_new_forkserver(NULL, st_pipe[0], ctl_pipe[1], fsrv->out_fd, fsrv->shm_coverage_map_shmid, fsrv->shm_fuzzing_shmid);
 
   /* PARENT PROCESS */
 
@@ -1212,11 +694,7 @@ void afl_fsrv_start(afl_forkserver_t *fsrv, char **argv,
 
     } else {
 
-      if (!fsrv->qemu_mode && !fsrv->cs_mode
-#ifdef __linux__
-          && !fsrv->nyx_mode
-#endif
-      ) {
+      if (!fsrv->qemu_mode && !fsrv->cs_mode) {
 
         WARNF(
             "Old fork server model is used by the target, this still works "
@@ -1665,10 +1143,6 @@ void afl_fsrv_kill(afl_forkserver_t *fsrv) {
   fsrv->fsrv_pid = -1;
   fsrv->child_pid = -1;
 
-#ifdef __linux__
-  afl_nyx_runner_kill(fsrv);
-#endif
-
 }
 
 /* Get the map size from the target forkserver */
@@ -1686,16 +1160,6 @@ u32 afl_fsrv_get_mapsize(afl_forkserver_t *fsrv, char **argv,
 void __attribute__((hot))
 afl_fsrv_write_to_testcase(afl_forkserver_t *fsrv, u8 *buf, size_t len) {
 
-#ifdef __linux__
-  if (unlikely(fsrv->nyx_mode)) {
-
-    fsrv->nyx_handlers->nyx_set_afl_input(fsrv->nyx_runner, buf, len);
-    return;
-
-  }
-
-#endif
-
 #ifdef AFL_PERSISTENT_RECORD
   if (unlikely(fsrv->persistent_record)) {
 
@@ -1815,81 +1279,13 @@ afl_fsrv_run_target(afl_forkserver_t *fsrv, u32 timeout,
   char             *persistent_out_fmt;
 #endif
 
-#ifdef __linux__
-  if (fsrv->nyx_mode) {
-
-    static uint32_t last_timeout_value = 0;
-
-    if (last_timeout_value != timeout) {
-
-      fsrv->nyx_handlers->nyx_option_set_timeout(
-          fsrv->nyx_runner, timeout / 1000, (timeout % 1000) * 1000);
-      fsrv->nyx_handlers->nyx_option_apply(fsrv->nyx_runner);
-      last_timeout_value = timeout;
-
-    }
-
-    enum NyxReturnValue ret_val =
-        fsrv->nyx_handlers->nyx_exec(fsrv->nyx_runner);
-
-    fsrv->total_execs++;
-
-    switch (ret_val) {
-
-      case Normal:
-        return FSRV_RUN_OK;
-      case Crash:
-      case Asan:
-        return FSRV_RUN_CRASH;
-      case Timeout:
-        return FSRV_RUN_TMOUT;
-      case InvalidWriteToPayload:
-        if (!!getenv("AFL_NYX_HANDLE_INVALID_WRITE")) { return FSRV_RUN_CRASH; }
-
-        /* ??? */
-        FATAL("FixMe: Nyx InvalidWriteToPayload handler is missing");
-        break;
-      case Abort:
-        FATAL("Error: Nyx abort occurred...");
-      case IoError:
-        if (*stop_soon_p) {
-
-          return 0;
-
-        } else {
-
-          FATAL("Error: QEMU-Nyx has died...");
-
-        }
-
-        break;
-      case Error:
-        FATAL("Error: Nyx runtime error has occurred...");
-        break;
-
-    }
-
-    return FSRV_RUN_OK;
-
-  }
-
-#endif
   /* After this memset, fsrv->trace_bits[] are effectively volatile, so we
      must prevent any earlier operations from venturing into that
      territory. */
 
-#ifdef __linux__
-  if (likely(!fsrv->nyx_mode)) {
-
-    memset(fsrv->trace_bits, 0, fsrv->map_size);
-    MEM_BARRIER();
 
-  }
-
-#else
   memset(fsrv->trace_bits, 0, fsrv->map_size);
   MEM_BARRIER();
-#endif
 
   /* we have the fork server (or faux server) up and running
   First, tell it if the previous run timed out. */
diff --git a/src/afl-fuzz-init.c b/src/afl-fuzz-init.c
index a2fc4528..609614ea 100644
--- a/src/afl-fuzz-init.c
+++ b/src/afl-fuzz-init.c
@@ -1273,14 +1273,7 @@ void perform_dry_run(afl_state_t *afl) {
         FATAL("Unable to execute target application ('%s')", afl->argv[0]);
 
       case FSRV_RUN_NOINST:
-#ifdef __linux__
-        if (afl->fsrv.nyx_mode && afl->fsrv.nyx_runner != NULL) {
-
-          afl->fsrv.nyx_handlers->nyx_shutdown(afl->fsrv.nyx_runner);
 
-        }
-
-#endif
         FATAL("No instrumentation detected");
 
       case FSRV_RUN_NOBITS:
diff --git a/src/afl-fuzz.c b/src/afl-fuzz.c
index 3480e04d..7bc3e2fd 100644
--- a/src/afl-fuzz.c
+++ b/src/afl-fuzz.c
@@ -434,50 +434,6 @@ static int stricmp(char const *a, char const *b) {
 
 }
 
-static void fasan_check_afl_preload(char *afl_preload) {
-
-  char   first_preload[PATH_MAX + 1] = {0};
-  char  *separator = strchr(afl_preload, ':');
-  size_t first_preload_len = PATH_MAX;
-  char  *basename;
-  char   clang_runtime_prefix[] = "libclang_rt.asan";
-
-  if (separator != NULL && (separator - afl_preload) < PATH_MAX) {
-
-    first_preload_len = separator - afl_preload;
-
-  }
-
-  strncpy(first_preload, afl_preload, first_preload_len);
-
-  basename = strrchr(first_preload, '/');
-  if (basename == NULL) {
-
-    basename = first_preload;
-
-  } else {
-
-    basename = basename + 1;
-
-  }
-
-  if (strncmp(basename, clang_runtime_prefix,
-              sizeof(clang_runtime_prefix) - 1) != 0) {
-
-    FATAL("Address Sanitizer DSO must be the first DSO in AFL_PRELOAD");
-
-  }
-
-  if (access(first_preload, R_OK) != 0) {
-
-    FATAL("Address Sanitizer DSO not found");
-
-  }
-
-  OKF("Found ASAN DSO: %s", first_preload);
-
-}
-
 /* Main entry point */

 int main(int argc, char **argv_orig, char **envp) {

@@ -1420,56 +1375,6 @@ int main(int argc, char **argv_orig, char **envp) {
 
   }
 
-  #ifdef __linux__
-  if (afl->fsrv.nyx_mode) {
-
-    if (afl->fsrv.nyx_standalone && strcmp(afl->sync_id, "default") != 0) {
-
-      FATAL(
-          "distributed fuzzing is not supported in this Nyx mode (use -Y "
-          "instead)");
-
-    }
-
-    if (!afl->fsrv.nyx_standalone) {
-
-      if (afl->is_main_node) {
-
-        if (strcmp("0", afl->sync_id) != 0) {
-
-          FATAL(
-              "for Nyx -Y mode, the Main (-M) parameter has to be set to 0 (-M "
-              "0)");
-
-        }
-
-        afl->fsrv.nyx_parent = true;
-        afl->fsrv.nyx_id = 0;
-
-      }
-
-      if (afl->is_secondary_node) {
-
-        long nyx_id = strtol(afl->sync_id, NULL, 10);
-
-        if (nyx_id == 0 || nyx_id == LONG_MAX) {
-
-          FATAL(
-              "for Nyx -Y mode, the Secondary (-S) parameter has to be a "
-              "numeric value and >= 1 (e.g. -S 1)");
-
-        }
-
-        afl->fsrv.nyx_id = nyx_id;
-
-      }
-
-    }
-
-  }
-
-  #endif
-
   if (afl->sync_id) {
 
     if (strlen(afl->sync_id) > 50) {
@@ -1742,28 +1647,8 @@ int main(int argc, char **argv_orig, char **envp) {
   afl->fsrv.use_fauxsrv = afl->non_instrumented_mode == 1 || afl->no_forkserver;
   afl->fsrv.max_length = afl->max_length;
 
-  #ifdef __linux__
-  if (!afl->fsrv.nyx_mode) {
-
-    check_crash_handling();
-    check_cpu_governor(afl);
-
-  } else {
-
-    u8 *libnyx_binary = find_afl_binary(argv[0], "libnyx.so");
-    afl->fsrv.nyx_handlers = afl_load_libnyx_plugin(libnyx_binary);
-    if (afl->fsrv.nyx_handlers == NULL) {
-
-      FATAL("failed to initialize libnyx.so...");
-
-    }
-
-  }
-
-  #else
   check_crash_handling();
   check_cpu_governor(afl);
-  #endif
 
   #ifdef __APPLE__
   setenv("DYLD_NO_PIE", "1", 0);
@@ -2100,6 +1985,9 @@ int main(int argc, char **argv_orig, char **envp) {
 
     }
 
+    afl->fsrv.shm_coverage_map_shmid = afl->shm.shm_id;
+    afl->fsrv.shm_fuzzing_shmid = afl->shm_fuzz->shm_id;
+
     u32 new_map_size = afl_fsrv_get_mapsize(
         &afl->fsrv, afl->argv, &afl->stop_soon, afl->afl_env.afl_debug_child);
 
